source/Locales/LawsonLocale/SharpElementsCoincideWithSpectralPoints.lagda

```diff
@@ -0,0 +1,761 @@
+   where
+    ♢ : is-basic Scott⦅𝓓⦆ 𝒦 (spectralᴰ-implies-directed-basisᴰ Scott⦅𝓓⦆ σᴰ) holds
+    ♢ = compact-opens-are-basic
+         Scott⦅𝓓⦆
+         (spectralᴰ-implies-directed-basisᴰ Scott⦅𝓓⦆ σᴰ)
+         𝒦
+         𝕜
+
+    lemma : (xs : List (index B𝓓)) → is-decidableₚ (x ∈ₛ βσ xs) holds
+    lemma []       = inr 𝟘-elim
+    lemma (i ∷ is) =
+     ∨-preserves-decidability pt (x ∈ₛ ↑ˢ[ βₖ i ]) (x ∈ₛ 𝜸 is) † IH
+      where
+       † : is-decidableₚ (x ∈ₛ ↑ˢ[ βₖ i ]) holds
+       † = 𝓈𝒽 (β i) (basis-is-compact i)
+
+       IH : is-decidableₚ (x ∈ₛ 𝜸 is) holds
+       IH = lemma is
+
+    ‡ : (xs : List (index B𝓓)) → βσ xs ＝ 𝒦 → is-decidableₚ (x ∈ₛ 𝒦) holds
+    ‡ xs p = transport (λ - → is-decidableₚ (x ∈ₛ -) holds) p (lemma xs)
+
+\end{code}
+
+The converse also holds meaning elements that admit decidable membership in
+compact Scott opens are _exactly_ the sharp elements.
+
+\begin{code}
+
+ admits-decidable-membership-in-compact-scott-opens-implies-is-sharp
+  : (x : ⟨ 𝓓 ⟩∙)
+  → admits-decidable-membership-in-compact-scott-opens x holds
+  → is-sharp x holds
+ admits-decidable-membership-in-compact-scott-opens-implies-is-sharp x φ c 𝕜 =
+  φ ↑ˢ[ (c , 𝕜) ] (principal-filter-is-compact₀ c 𝕜)
+
+ characterization-of-sharp-elements
+  : (x : ⟨ 𝓓 ⟩∙)
+  → (admits-decidable-membership-in-compact-scott-opens x ⇔ is-sharp x) holds
+ characterization-of-sharp-elements x = † , ‡
+  where
+   † = admits-decidable-membership-in-compact-scott-opens-implies-is-sharp x
+   ‡ = sharp-implies-admits-decidable-membership-in-compact-scott-opens x
+
+\end{code}
+
+\subsection{A corollary of the characterization}
+
+Because clopens are compact in compact frames, we can also prove that admitting
+decidable membership in Scott clopens is a necessary condition for an element of
+the domain to be sharp.
+
+We do not need this result for the main result in this module, but we note it
+down regardless as it is a potentially useful observation.
+
+\begin{code}
+
+ admits-decidable-membership-in-scott-clopens : ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+ admits-decidable-membership-in-scott-clopens x =
+  Ɐ 𝒦 ꞉ ⟨ 𝒪 Scott⦅𝓓⦆ ⟩ , is-clopen (𝒪 Scott⦅𝓓⦆) 𝒦 ⇒ is-decidableₚ (x ∈ₛ 𝒦)
+
+ sharp-implies-admits-decidable-membership-in-scott-clopens
+  : (x : ⟨ 𝓓 ⟩∙)
+  → is-sharp x holds
+  → admits-decidable-membership-in-scott-clopens x holds
+ sharp-implies-admits-decidable-membership-in-scott-clopens x 𝓈𝒽 K χ =
+  ψ K κ
+   where
+    ψ : admits-decidable-membership-in-compact-scott-opens x holds
+    ψ = sharp-implies-admits-decidable-membership-in-compact-scott-opens x 𝓈𝒽
+
+    κ : is-compact-open Scott⦅𝓓⦆ K holds
+    κ = clopens-are-compact-in-compact-frames
+         (𝒪 Scott⦅𝓓⦆)
+         Scott⦅𝓓⦆-is-compact
+         K
+         χ
+
+\end{code}
+
+What can be said about the converse of this implication? In other words, what is
+the meaning of the set of elements that admit decidable membership in Scott
+clopens. I do not know the answer yet.
+
+TODO: think more about this.
+
```

Comment:
Don't *all* element admit decidability in Scott clopens? You first test whether bottom is in the set. If it is, then also your element is. Otherwise the set must be empty, and so your element is not there.

Reply:
Yes. This property of having decidable membership in Scott clopens doesn’t mean anything since the Scott topology is connected. I will update the remark here to explain this.

The interesting version of this question is: what can be said about elements that admit decidable membership in the clopens of Patch(Scott(D))? I will keep thinking about this.